datastream/tools/split_dataframe.py

```diff
@@ -0,0 +1,155 @@
+from __future__ import annotations
+from typing import Tuple, Union, Dict, Optional
+from pathlib import Path
+import numpy as np
+import pandas as pd
+
+from datastream import json
+
+
+def split_dataframe(
+    dataframe: pd.DataFrame,
+    key_column: str,
+    proportions: Dict[str, float],
+    filepath: Optional[Path] = None,
+    stratify_column: Optional[str] = None,
+):
+    '''
+    Split and save result. Add new examples and continue from the old split.
+
+    As new examples come in it can handle:
+    - Changing test size
+    - Adapt after removing examples from dataset
+    - Adapt to new stratification
+    '''
+    if abs(sum(proportions.values()) - 1.0) >= 1e-6:
+        raise AssertionError('Expected sum of proportions to be 1')
+    
+    if filepath is not None and filepath.exists():
+        split = json.read(filepath)
+
+        if set(proportions.keys()) != set(split.keys()):
+            raise AssertionError(
+                'Expected split names in split file to be the same as the keys'
+                'of proportions'
+            )
+    else:
+        split = {
+            split_name: list()
+            for split_name in proportions.keys()
+        }
+
+    split_proportions = tuple(proportions.items())
+
+    for split_name, proportion in split_proportions[:-1]:
+        if stratify_column is None:
+            split = split_proportion(
+                dataframe[key_column],
+                proportion,
+                split_name,
+                split,
+            )
+        else:
+            for strata in stratas(dataframe, stratify_column):
+                split = split_proportion(
+                    dataframe[key_column],
```

Comment:
Should be `dataframe[dataframe[stratify_column] == strata]` or am I mistaken?

Reply:
Yes! This is the bug I found when I was resolving your comment above :+1: